### DISCUSSION TITLE
descriptions for custom types

### DISCUSSION THREAD

**Discussion by ActualMasterOogway** [JohnnyMorganz/luau-lsp#627] [Q&A] [answered]
How can i add a description in my type definition?
so it would look something like this: 
![image](https://github.com/JohnnyMorganz/luau-lsp/assets/125769645/8922e593-563a-4718-b67c-fbf7ca18c744)

**Comment by JohnnyMorganz** [comment, marked as answer]
Unfortunately it is not very well documented, but you can use a documentations file alongside a definitions file.

You can find an example of such a file at https://raw.githubusercontent.com/MaximumADHD/Roblox-Client-Tracker/roblox/api-docs/en-us.json which is a `DocumentationDatabase` in JSON format.

To understand its structure, see https://github.com/luau-lang/luau/blob/master/Analysis/include/Luau/Documentation.h

I want to make this simpler using doc-comments inlined in source code instead. This is currently missing some Luau features to make possible. Relevant issue: #271
